.travis.yml

```diff
@@ -1,29 +1,94 @@
 language: cpp
 dist: trusty
-sudo: required # the doc target uses sudo to install dependencies
+sudo: false
+
+os: linux
+
+git:
+  depth: 1
 
-os:
-  - linux
-  - osx
 
 env:
   global:
     - secure: |-
         a1eovNn4uol9won7ghr67eD3/59oeESN+G9bWE+ecI1V6yRseG9whniGhIpC/YfMW/Qz5I
         5sxSmFjaw9bxCISNwUIrL1O5x2AmRYTnFcXk4dFsUvlZg+WeF/aKyBYCNRM8C2ndbBmtAO
         o1F2EwFbiso0EmtzhAPs19ujiVxkLn4=
-  matrix:
-    - BUILD=Doc
-    - BUILD=Debug STANDARD=14
-    - BUILD=Release STANDARD=14
 
 matrix:
-  exclude:
-    - os: osx
-      env: BUILD=Doc
   include:
+      # Documentation
+    - env: BUILD=Doc
+      sudo: required
+      # g++ 6 on Linux with C++14
+    - env: CXX_COMPILER=g++-6 BUILD=Debug STANDARD=14
```

Comment:
Why not use `CXX` instead of `CXX_COMPILER` and simplify `before_script`?

Reply:
Travis exports `CXX` as `g++` in between of `matrix` and `before_script`, so the value set in `matrix` would be lost.